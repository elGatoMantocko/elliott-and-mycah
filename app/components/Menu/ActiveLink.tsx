import Link, { LinkProps } from '@material-ui/core/Link';
import { makeStyles } from '@material-ui/core/styles';
import * as React from 'react';
import { useHistory, useLocation } from 'react-router';

const useLinkColorStyles = makeStyles((theme) => {
  const { pathname } = useLocation();
  return {
    root: {
      cursor: ({ disabled }: { disabled?: boolean }) => (disabled === true ? 'default' : 'pointer'),
      color: ({ disabled, href }: { disabled?: boolean; href?: string }) =>
        disabled
          ? theme.palette.text.disabled
          : theme.palette[pathname === href ? 'primary' : 'secondary'].main,
    },
  };
});

type ActiveLinkProps = {
  disabled?: boolean;
} & LinkProps;
export const ActiveLink = ({ disabled = false, ...linkProps }: ActiveLinkProps) => {
  const { push } = useHistory();
  return (
    <Link
      {...linkProps}
      href={undefined}
      classes={useLinkColorStyles({ disabled, href: linkProps.href })}
      underline={disabled ? 'none' : undefined}
      onClick={(e: React.MouseEvent) => {
        if (disabled) {
          return e.preventDefault();
        }
        linkProps.href != null && push(linkProps.href);
      }}
    />
  );
};
