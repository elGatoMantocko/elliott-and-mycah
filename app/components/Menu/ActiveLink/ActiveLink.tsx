import { Link } from '@mui/material';
import React from 'react';
import { useLocation } from 'react-router';
import { Link as RouterLink, LinkProps } from 'react-router-dom';

type ActiveLinkProps = {
  disabled?: boolean;
} & LinkProps;
export const ActiveLink = ({ disabled = false, ...linkProps }: ActiveLinkProps) => {
  const { pathname } = useLocation();

  return (
    <Link
      {...linkProps}
      component={RouterLink}
      sx={{ cursor: disabled ? 'default' : 'pointer' }}
      color={disabled ? 'textSecondary' : pathname === linkProps.to ? 'primary' : 'secondary'}
      underline={disabled ? 'none' : undefined}
    />
  );
};
