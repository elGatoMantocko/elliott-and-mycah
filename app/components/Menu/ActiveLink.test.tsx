import { render, screen, waitFor } from '@testing-library/react';
import { expect } from 'chai';
import * as React from 'react';
import { MemoryRouter } from 'react-router-dom';

import { ActiveLink } from './ActiveLink';

describe('<ActiveLink>', () => {
  it('should render a link', async () => {
    render(<ActiveLink href="/test">Some test link</ActiveLink>, { wrapper: MemoryRouter });

    const link = await waitFor(() => screen.getByText('Some test link'));
    expect(link).to.exist;
    expect(link.hasAttribute('href')).to.be.false;
  });
});
