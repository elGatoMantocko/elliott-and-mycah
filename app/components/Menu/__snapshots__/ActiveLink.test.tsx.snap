// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActiveLink> should render a link 1`] = `
<div>
  <a
    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1cep7cv-MuiTypography-root-MuiLink-root"
  >
    Some test link
  </a>
</div>
`;

exports[`<ActiveLink> should render active on the current path 1`] = `
<div>
  <a
    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-12gn65i-MuiTypography-root-MuiLink-root"
  >
    Some test link
  </a>
</div>
`;

exports[`<ActiveLink> should render disabled link 1`] = `
<div>
  <a
    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-yh9o1u-MuiTypography-root-MuiLink-root"
  >
    Some test link
  </a>
</div>
`;
